vsg/__main__.py

```diff
@@ -393,13 +409,21 @@ def main():
 
     dStyle = read_predefined_style(commandLineArguments.style)
 
-    configuration = read_configuration_files(dStyle, commandLineArguments)
+    fix_only = settings.open_configuration_file(commandLineArguments.fix_only)
+
+    if commandLineArguments.configuration:
+        configuration = read_configuration_files(dStyle, commandLineArguments)
+
+        dIndent = read_indent_configuration(configuration)
 
-    dIndent = read_indent_configuration(configuration)
+        update_command_line_arguments(commandLineArguments, configuration)
+
+    else:
+        update_command_line_arguments2(commandLineArguments)
 
-    fix_only = open_fix_file(commandLineArguments.fix_only)
+        configuration = read_configuration_files2(dStyle, commandLineArguments)
 
-    update_command_line_arguments(commandLineArguments, configuration)
+        dIndent = read_indent_configuration(configuration)
 
     configuration = add_debug_to_configuration(commandLineArguments, configuration)
```

Comment:
I think this could be better done with [logging](https://docs.python.org/3/howto/logging.html) instead of putting debug in the configuration dictionary, even if you want to be able to enable debugging from the configuration.

Reply:
I agree.